### ISSUE TITLE
request from network (not localhost) fails with error 500

### ISSUE THREAD

**Issue by unclesamwk** [josh-berry/cronitor#1]
Hey,

i've an issue.
I create an docker container and run cronitor in it.
It works. I can create cronitor-run successfully and dashboard display it.
If i create new task from network, the dashboard fails with error 500.
The logfile creates successfully.

Debug output:
cronitor    | rnado.application:Uncaught exception GET / (172.17.0.1)
cronitor    | HTTPServerRequest(protocol='http', host='127.0.0.1:8434', method='GET', uri='/', version='HTTP/1.1', remote_ip='172.17.0.1', headers={'Accept-Language': 'en-US,en;q=0.5', 'Accept-Encoding': 'gzip, deflate', 'Host': '127.0.0.1:8434', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'User-Agent': 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:53.0) Gecko/20100101 Firefox/53.0', 'Connection': 'keep-alive', 'Cookie': 'grav-tabs-state={%22tab-content.options.advanced%22:%22data.advanced%22%2C%22tab-content.options%22:%22data.options%22}; rack.session=BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiRTk4Njc3YWFmMmNjNTk1OGRkNzcy%0AZmMzOTEzOTY0MmNhMjIwNGFmNDgzZjYxOTBhMjllOTA0NjgxZDFmODI0NjUG%0AOwBGSSIJY3NyZgY7AEZJIiVkYTFlZDYyZWZjODI5YmEyMmUxNzI2ZjI5ZTNm%0AYmM0YgY7AEZJIg10cmFja2luZwY7AEZ7B0kiFEhUVFBfVVNFUl9BR0VOVAY7%0AAFRJIi1hOGU3N2E3MDgzMzc3ZWE2Mjc2ODliYjM0OWZhNTRkZmEyNmQyNThi%0ABjsARkkiGUhUVFBfQUNDRVBUX0xBTkdVQUdFBjsAVEkiLWRkMDY1ZWQyNjNj%0ANjdkNzk5Zjk0M2FiNmMzOWI1NWM1ZTAwOGNiYjUGOwBGSSIKZmxhc2gGOwBG%0AewA%3D%0A--1159726ce6d991d052749b088fef3b286b357293; minicron.session=BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiRTk4Njc3YWFmMmNjNTk1OGRkNzcy%0AZmMzOTEzOTY0MmNhMjIwNGFmNDgzZjYxOTBhMjllOTA0NjgxZDFmODI0NjUG%0AOwBGSSIJY3NyZgY7AEZJIiVkYTFlZDYyZWZjODI5YmEyMmUxNzI2ZjI5ZTNm%0AYmM0YgY7AEZJIg10cmFja2luZwY7AEZ7B0kiFEhUVFBfVVNFUl9BR0VOVAY7%0AAFRJIi1hOGU3N2E3MDgzMzc3ZWE2Mjc2ODliYjM0OWZhNTRkZmEyNmQyNThi%0ABjsARkkiGUhUVFBfQUNDRVBUX0xBTkdVQUdFBjsAVEkiLWRkMDY1ZWQyNjNj%0ANjdkNzk5Zjk0M2FiNmMzOWI1NWM1ZTAwOGNiYjUGOwBGSSIKZmxhc2gGOwBG%0AewA%3D%0A--0be11236a85db32bc11109dce2e2b94e7144ed40', 'Cache-Control': 'max-age=0', 'If-None-Match': '"0ad0f0473fe750e5a3a785b4efd74e98864e1272"', 'Upgrade-Insecure-Requests': '1'})
cronitor    | Traceback (most recent call last):
cronitor    |   File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 1509, in _execute
cronitor    |     result = method(*self.path_args, **self.path_kwargs)
cronitor    |   File "/opt/cronitor/cronitor-server", line 77, in get
cronitor    |     self.render("index.html", jobs=jobset)
cronitor    |   File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 724, in render
cronitor    |     html = self.render_string(template_name, **kwargs)
cronitor    |   File "/usr/local/lib/python2.7/dist-packages/tornado/web.py", line 865, in render_string
cronitor    |     return t.generate(**namespace)
cronitor    |   File "/usr/local/lib/python2.7/dist-packages/tornado/template.py", line 348, in generate
cronitor    |     return execute()
cronitor    |   File "index_html.generated.py", line 9, in _tt_execute
cronitor    |     _tt_tmp = job.status  # index.html:17
cronitor    |   File "/opt/cronitor/cronitor/model.py", line 133, in status
cronitor    |     return self.latest_entry.status
cronitor    |   File "/opt/cronitor/cronitor/model.py", line 167, in status
cronitor    |     if self.rc != 0:
cronitor    |   File "/opt/cronitor/cronitor/model.py", line 191, in rc
cronitor    |     self._read()
cronitor    |   File "/opt/cronitor/cronitor/model.py", line 155, in _read
cronitor    |     assert m
cronitor    | AssertionError
cronitor    | ERROR:tornado.access:500 GET / (172.17.0.1) 18.28ms

Any idea?

greetings Sam

**Comment by josh-berry** [comment]
Hi, unfortunately Cronitor could really use some better error handling, and I apologize for the trouble you're having.

It looks like there was a problem on the server side, based on the last line of your cronitor-run log:

```
cronitor | ERROR:tornado.access:500 GET / (172.17.0.1) 18.28ms
```

The "500" is an HTTP status code for "Internal Server Error". Do you have any server-side logs you can share?

**Comment by josh-berry** [comment]
Sorry, ignore my previous comment -- the HTTP request at the top looked like it was client output.

You may have a corrupt log entry. All of the logs are saved in plain text files, in the log directory. If you don't have a lot of logs, you can look thru the files to see if you can find the problematic entry.

Entries are supposed to be formatted like:

```
Var: value
Var: value
...
[one blank line]
[log text]
```

For example:

```
Host: tux
ENV.LOGNAME: root
Command: /srv/admin/bin/btrfs-scrub /mnt/fsroot
ENV.DR_LIVE_VOL: /mnt/fsroot/@srv
ENV.PWD: /root
Directory: /root
ENV.SHELL: /bin/bash
ENV.PATH: /bin:/usr/bin:/sbin:/usr/sbin:/usr/local/bin:/usr/local/sbin

[log text]
```

If the server got interrupted writing a log entry, or otherwise somehow managed to write a bogus entry into that directory, you might see this kind of issue.

**Comment by unclesamwk** [comment]
Hey,

yeah its really an problem with the command i use with cronitor-run. sorry adn thx for your answer.

I change the template and others ... may i do an push?

greetings
Sam

**Comment by josh-berry** [comment]
Send me a pull request and I can take a look!

**Comment by josh-berry** [comment]
I've added a bunch of error handling to address the most common cases like the original issue; now, if Cronitor encounters a bad log file, it will just try to display as much as it can, rather than throwing Python exceptions / dumping backtraces. Hopefully this will at least make it easier to troubleshoot in future.

Closing this issue, but we can keep looking at your template changes in the pull request (thanks again for sending it!).
